warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/WhindexIngest.java

```diff
@@ -0,0 +1,212 @@
+     * {@code OverloadedWhindexField}s are source fields ("{@code SRC_FIELD}") that become redundant once whindex entries are generated. They are marked to be
+     * removed.
+     *
+     * @param field
+     *            the field to check.
+     * @return {@code true} if {@code field} is an {@code OverloadedWhindexField} (marked for removal)
+     */
+    boolean isOverloadedWhindexField(String field);
+
+    /**
+     * // todo Given a "{@code RULE}", return a {@code Multimap<String, NormalizedContentInterface>} of whindex fields ("{@code DST_FIELD}") mapped to the
+     * values specified by the {@code RULE}.
+     *
+     * @return the mapping of whindex fields to values.
+     */
+    Multimap<String,NormalizedContentInterface> getWhindexFields(Multimap<String,NormalizedContentInterface> eventFields);
+
+    /**
+     * Responsible for parsing the {@code .rules} passed to the {@link WhindexIngest} and holding configuration information.
+     */
+    class WhindexFieldNormalizer {
+
+        // The property name that holds the whindex rules string in the configuration xml
+        public static final String RULE = "rules";
+
+        // The delimiter used to separate distinct RULEs within the <type>.RULES property
+        // ie: RULE;RULE;RULE;...
+        public static final String RULE_DELIMITER = ";";
+
+        // The delimiter used to separate the FIELDS side of a RULE from the VALUE,VALUE,... side of a RULE.
+        public static final String FIELDS_VALUES_DELIMITER = "::";
+
+        // The delimiter used to separate distinct FIELDs and VALUEs within a RULE.
+        // ie: FIELD,FIELD,FIELD::VALUE,VALUE,VALUE
+        // The left side is shorthanded (?) to FIELDS::VALUE,VALUE,VALUE,... which omits the fact that FIELDS is made up of several FIELDs.
+        public static final String FIELD_VALUE_DELIMITER = ",";
+
+        // The delimiter used to separate SRC_FIELD:DST_FIELD:DELETE_SRC within the <type>.FIELDS.<some_field> property
+        public static final String SRC_DST_DEL_DELIMITER = ":";
+
+        // Mappings of source fields to their respective whindex field.
+        private final Multimap<String,String> whindexFieldDefinitions = LinkedListMultimap.create();
+        private final Set<String> overloadedFields = new HashSet<>();
+        private MarkingFunctions markingFunctions;
+
+        /**
+         * Parses the {@code config}'s "{@code <datatype>.rules"} property, generating whindex entries based on each rule.
+         *
+         * @param type
+         *            the datatype we're looking for.
+         * @param config
+         *            the config instance that holds the rules.
+         */
+        public void setup(Type type, Configuration config) {
+            markingFunctions = MarkingFunctionsFactory.createMarkingFunctions();
+
+            String whindexRules = config.get(type.typeName() + "." + RULE, "");
+            if (whindexRules.isEmpty()) {
+                return;
+            }
+
```

Comment:
We are missing configuration (1) from the description above; namely the fields from which those values apply.

Reply:
We need to be able to create multiple whindex configuration

Reply:
How should we expect the different whindex configurations to be distinguished? Will they have an instance value in their configuration property, or will we have multiple configurations mapped to the same `<type>.rules` property?

Reply:
For example, do we need to support something akin to 
```
wiki.rules.1=`...`
wiki.rules.2=`...`
```
or something different?